src/base/string_buffer.hpp

```diff
@@ -108,13 +108,21 @@ void string_buffer::operator+= (const char ch)
 {
   if (dim < m_len + 2)
```

Comment:
Why "dim < m_len + 2" and not "dim <= m_len"? The "2" seems to be required only if dim == 0.

Reply:
m_len doesn't count the ending '\0'
for the case when dim != 0 and dim == m_len+1 and I want to add another char then I have to realloc => condition to realloc is dim <= m_len+1 <=> dim < m_len+2
I'l rewrite the conditions as:
if(dim == 0)
  extend(2);
else if(dim <= m_len+1)
  extend(1);

Reply:
Ok, looks good. Please add a comment to mention that m_len does not include '\0'.